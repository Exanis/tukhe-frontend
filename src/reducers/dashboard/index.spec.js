import slice, { selectDashboard } from './index';

it('Should correctly use reducers', () => {
    const state = {
        dashboards: [],
        currentDashboard: null,
        widgets: [],
        data: {}
    };
    const action = {
        type: slice.actions.setDashboards,
        payload: ['test', 'test2']
    };

    expect(slice.reducer(state, action)).toMatchSnapshot();

    action.type = slice.actions.setWidgets;
    expect(slice.reducer(state, action)).toMatchSnapshot();

    action.type = slice.actions.selectDashboard;
    action.payload = 'test';
    expect(slice.reducer(state, action)).toMatchSnapshot();

    action.type = slice.actions.setData;
    action.payload = {
        widget: 'test',
        data: 'value'
    };
    expect(slice.reducer(state, action)).toMatchSnapshot();

    action.type = slice.actions.updateWidgetData;
    action.payload = {
        widget: 'test',
        select: el => el === 4,
        update: el => el + 2
    };
    state.data['test'] = [4, 5];
    expect(slice.reducer(state, action)).toMatchSnapshot();

    action.type = slice.actions.addToWidgetData;
    action.payload = {
        widget: 'test',
        elements: [1, 2, 3]
    };
    state.data['test'] = [4, 5];
    expect(slice.reducer(state, action)).toMatchSnapshot();
});

it('Should correctly select dashboards', () => {
    const state = {
        dashboard: 'test',
        error: 'bad'
    };

    expect(selectDashboard(state)).toEqual('test');
});
