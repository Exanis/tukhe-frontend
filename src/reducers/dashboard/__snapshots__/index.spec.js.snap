// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly use reducers 1`] = `
Object {
  "currentDashboard": null,
  "dashboards": Array [
    "test",
    "test2",
  ],
  "data": Object {},
  "widgets": Array [],
}
`;

exports[`Should correctly use reducers 2`] = `
Object {
  "currentDashboard": null,
  "dashboards": Array [],
  "data": Object {},
  "widgets": Array [
    "test",
    "test2",
  ],
}
`;

exports[`Should correctly use reducers 3`] = `
Object {
  "currentDashboard": "test",
  "dashboards": Array [],
  "data": Object {},
  "widgets": Array [],
}
`;

exports[`Should correctly use reducers 4`] = `
Object {
  "currentDashboard": null,
  "dashboards": Array [],
  "data": Object {
    "test": "value",
  },
  "widgets": Array [],
}
`;

exports[`Should correctly use reducers 5`] = `
Object {
  "currentDashboard": null,
  "dashboards": Array [],
  "data": Object {
    "test": Array [
      6,
      5,
    ],
  },
  "widgets": Array [],
}
`;

exports[`Should correctly use reducers 6`] = `
Object {
  "currentDashboard": null,
  "dashboards": Array [],
  "data": Object {
    "test": Array [
      1,
      2,
      3,
      4,
      5,
    ],
  },
  "widgets": Array [],
}
`;
